kolibri/plugins/management/assets/src/vue/modal/index.vue

```diff
@@ -1,66 +1,115 @@
 <template>
 
-  <div class="modal-root" v-on:keyup.esc="closeModal">
-    <div class="modal" v-if="showModal" transition="modal">
-      <div class="modal-wrapper">
-        <div class="modal-container">
-          <button @click="closeModal" class="close-btn">
-            <svg src="../icons/close.svg"></svg>
-            <span class="visuallyhidden">Close</span>
-          </button>
-          <div class="modal-header">
-            <slot name="header">
-              Kolibri
-            </slot>
-          </div>
-          <div class="modal-body">
-            <slot name="body">
-              disappear in oblivion..
-            </slot>
-          </div>
-          <div class="modal-footer">
-            <slot name="footer">
-              <button @click="closeModal" class="close-btn">OK</button>
-            </slot>
-          </div>
-        </div>
-      </div>
-    </div>
-
-    <div @click="openModal">
-    <!-- wrap this named slot so that the openModal method logic is encapsulated inside this modal component, but the parent component can pass anything to this slot for styling purpose -->
-      <slot name="openbtn">
-        <button>{{ btntext }}</button>
+  <!-- Accessibility properties for the overlay -->
+
+  <!-- Aria-Hidden and TabIndex in .modal might not be necessary because of conditional rendering -->
+  <!-- mostly there in case we switch to v-show -->
+  <div class="modal-overlay"
+    v-if="visible"
+    @keyup.esc="closeModal"
+    @click="bgClick($event)"
+    v-el:modal-overlay
+    id="modal-window"
+    role="dialog"
+    :aria-hidden="(!visible).toString()"
```

Comment:
> mostly there in case we switch to v-show

I think in that case it's probably not necessary. Choosing `v-if` to remove things from the DOM seems legit.

Are you thinking of a specific scenario where we might want to switch to `v-show`?